### ISSUE TITLE
🐛 ZX-Checker produces invalid result

### ISSUE THREAD

**Issue by alan23273850** [cda-tum/qcec#200]
### mqt.qcec version

`mqt.qcec>=2.0.0`

### OS

Ubuntu 20.04.5 LTS

### Python version

3.8.10

### C++ compiler

g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

### Description

The ZX Checker yields incorrect results in certain cases, e.g., the following circuits
```qasm
OPENQASM 2.0;
include "qelib1.inc";
qreg qubits[1];
```
and
```qasm
OPENQASM 2.0;
include "qelib1.inc";
qreg qubits[1];
h qubits[0];
```
are erroneously considered equivalent.

### Expected behavior

Circuits should be detected as non-equivalent.

### How to Reproduce

```python
from mqt import qcec
from qiskit import QuantumCircuit

qc1 = QuantumCircuit(1)

qc2 = QuantumCircuit(1)
qc2.h(0)

result = qcec.verify(qc1, qc2, run_alternating_checker=False, run_simulation_checker=False)
print(result.equivalence)
```

**Comment by burgholzer** [comment]
@alan23273850 Thanks for reporting this bug.

I was able to reproduce the issue and took the liberty to reduce the complexity of the example you provided to a minimal reproducing example. I correspondingly updated the description in the PR. The culprit here is the ZX Checker.

@pehamTom could you please take a look?

**Comment by pehamTom** [comment]
Should be a quick fix. Gonna take care of that right now.

**Comment by burgholzer** [comment]
Fixed as of version 2.1.1. Just released the new version. Should be available within the next hour.
